### ISSUE TITLE
decide git branching model

### ISSUE THREAD

**Issue by muatik** [SWE574-RenameMe/swe574-CoolName#2]
First of all, we need to decide how we are going to use git. There are many ways, the most simple one is that everybody has a clone of this repo, with only one branch named `master`. And all work is done on this branch, and the changes are sent as a pull request to the organization repo. However, there are other models. 

For instance, one of the most used models is the feature-branching model. http://nvie.com/posts/a-successful-git-branching-model/ I suggest following this model. Let's discuss.

if you accept this model thumbs up, please.

**Comment by fkucuk** [comment]
We use feature-branching model at work. It is very convenient. It categorizes your development into features by both physically and conceptually. Multiple features can be developed simultaneously. And each deployment you know what features are included. It is also useful for deployment documentation.

**Comment by TalatCikikci** [comment]
The branching structure defined in the link you have provided is what we are using at my workplace. I find it very useful as it allows the code to be more manageable in smaller scales.

**Comment by muatik** [comment]
Since both the group name and repository name have been decided, should we fork this repo? From now own, we should not make any changes on this repo directly, except issues and wiki pages. 

If you have not yet read the url I provided, please give priority to it. It explains how we are going to use git system. If you have any question, do not hesitate to ask.

**Comment by tokonu** [comment]
@muatik should we create a separate branch for mobile-frontend-backend or use a dev branch for all?

**Comment by muatik** [comment]
we can use the same dev branch for all. since they share nothing, any conflict will not occur.

**Comment by fkucuk** [comment]
On 25th March we've talked about forking the main repo in our own repositories. I was trying to setup my development environment according to wiki contribution page. I've had forked the repo very very long time ago and I've realized that my version of repo is outdated. So I had to synchronize it with the main repo. I've found [this](https://help.github.com/articles/syncing-a-fork/) walk through in github's help page. 

Is this the right way to keep our copy of repo  updated?
